sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala

```diff
@@ -29,16 +30,19 @@ import org.apache.spark.sql.sources.Filter
 import org.apache.spark.sql.v2.avro.AvroScan
 import org.apache.spark.util.{SerializableConfiguration, Utils}
 
-object ExternalSource {
+object ExternalSource extends Logging {
+  val avroScanClassName = "org.apache.spark.sql.v2.avro.AvroScan"
 
   lazy val hasSparkAvroJar = {
-    val loader = Utils.getContextOrSparkClassLoader
-
     /** spark-avro is an optional package for Spark, so the RAPIDS Accelerator
      * must run successfully without it. */
-    Try(loader.loadClass("org.apache.spark.sql.v2.avro.AvroScan")) match {
-      case Failure(_) => false
-      case Success(_) => true
+    Utils.classIsLoadable(avroScanClassName) && {
+      Try(ShimLoader.loadClass(avroScanClassName)).map(_ => true)
+        .getOrElse {
+          logWarning("WARNING: Avro library not found by the RAPIDS plugin. If needed, please use " +
```

Comment:
Address PR checks

scalastyle check flagged on the PR 
```
error file=/home/runner/work/spark-rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala message=File line length exceeds 100 characters line=42
```

Previous comment not addressed yet
https://github.com/NVIDIA/spark-rapids/pull/5716#discussion_r889219448